Alicerce/Sources/Network/Network.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  Network.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 06/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public enum Network {
+
+    // MARK: - TypeAlias
+
+    public typealias CompletionClosure = (_ inner: () throws -> Data) -> Void
+
+    // MARK: - Network Error
+
+    enum Error: Swift.Error {
+        case http(code: HTTP.StatusCode, description: String?)
+        case noData
+        case url(Swift.Error)
```

Comment:
Not sure about naming this error case `url`, but can't think of a nicer alternative... 🤔

The first thing that comes to mind is `network` 😆, and these errors will be stuff like disconnected, timeout, etc